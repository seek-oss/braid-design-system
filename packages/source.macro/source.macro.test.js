// eslint-disable-next-line import/no-unresolved
const { default: pluginTester } = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  pluginName: 'source.macro',
  snapshot: true,
  tests: {
    'smoke test': {
      babelOptions: { filename: __filename },
      code: `
        import source from './source.macro';

        const result = source({
          // comment
          hello: () => 'world'
        });
      `,
    },
    'ignores types': {
      babelOptions: { filename: __filename },
      code: `
        import { Source } from './source.macro';

        const foo = 'bar';
      `,
    },
  },
});
