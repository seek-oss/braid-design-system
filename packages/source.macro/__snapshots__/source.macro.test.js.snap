// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source.macro ignores types: ignores types 1`] = `

import { Source } from './source.macro';

const foo = 'bar';

      ↓ ↓ ↓ ↓ ↓ ↓

const foo = 'bar';


`;

exports[`source.macro smoke test: smoke test 1`] = `

import source from './source.macro';

const result = source({
  // comment
  hello: () => 'world'
});

      ↓ ↓ ↓ ↓ ↓ ↓

const result = {
  code: "{\\n  // comment\\n  hello: () => 'world'\\n}",
  value: {
    // comment
    hello: () => 'world',
  },
};


`;
